### ISSUE TITLE
Doubts about masked_scatter gradient

### ISSUE THREAD

**Issue by BBuf** [pytorch/pytorch#66360]
## 🐛 Bug

When I run a sample of the `masked_scatter` operator, I found that the input gradient does not seem to meet my expectations. 

## To Reproduce

```python
import torch

dest = torch.tensor([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], dtype=torch.float32)

mask = torch.tensor((0, 0, 0, 0, 1, 0, 1, 0, 1, 0), dtype=torch.bool)

src = torch.tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=torch.float32)
src.requires_grad = True

dest = dest.masked_scatter(mask, src)

out = dest.sum()
out.backward()

print(src.grad)
```

The printed result is: 

```shell
tensor([1., 1., 1., 0., 0., 0., 0., 0., 0., 0.])
```

But according to the semantics of masked_scatter, I think the gradient should be：

```shell
tensor([0., 0., 0., 0., 1., 0., 1., 0., 1., 0.])
```

Because only the position with the value of `1` in the mask contributes to the output。

Hope to get answers, thank you！

## Environment

- Pytorch 1.9.1
